docs(sql): separating entity definition (#1172)
nestjs/docs.nestjs.com: content/techniques/sql.md

> warning **Warning** Note that entities that aren't registered through the `forFeature()` method, but are only referenced from the entity (via a relationship), won't be included by way of the `autoLoadEntities` setting.

#### Separating Entity Definition

You can define an entity and its columns right in the model, using decorators. But some people prefer to define an entity and its columns inside separate files which are called ["entity schemas"](https://typeorm.io/#/separating-entity-definition).

```typescript
import { EntitySchema } from 'typeorm';
import { User } from './user.entity';

export const UserSchema = new EntitySchema<User>({
    name: 'User',
    target: User,
    columns: {
        id: {
            type: Number,
            primary: true,
            generated: true
        },
        firstName: {
            type: String
        },
        lastName: {
            type: String
        },
        isActive: {
            type: Boolean,
            default: true
        }
    },
    relations: {
        photos: {
            type: 'one-to-many',
            target: 'Photo' // The name of the PhotoSchema
        }
    }
});
```

> > error **Warning** If you provide the `target` option the `name` option has to be the same as the name of the target class.

Reviewer: ```suggestion
> error **Warning** If you provide the `target` option, the `name` option value has to be the same as the name of the target class.
```

> > If you not provide the target you can use any name.

Reviewer: ```suggestion
> If you do not provide the `target` you can use any name.
```

> Nest allows you to use EntitySchema instance where Entity is expected, for example:

Reviewer: ```suggestion
Nest allows you to use an `EntitySchema` instance wherever an `Entity` is expected, for example:
```
